### PR TITLE
dotsize reflects repeat gps updates

### PR DESCRIPTION
**Pull request by owades**
- GPS update dot size increases by 20% of original size for each repeat update sharing same exact latlong. Did this by replacing List<GPSData> with Map<String,GPSData>, which maps each latLong combo with the corresponding GPSData. Increment when already exists.
- Fixes bug where DesertRoadrunner trip names weren't appearing

Potential Issues:
- TrainingDataUtil.java changes remain untested. Will seek guidance on how to test.

### CODE REVIEW

File: gtfu/src/main/java/gtfu/GPSData.java [hunk 1]
```diff
@@ -4,11 +4,13 @@
     public long millis;
     public float lat;
     public float lon;
+    public int count;
 
-    public GPSData(long millis, float lat, float lon) {
+    public GPSData(long millis, float lat, float lon, int count) {
```
Comment (owades) [inline gtfu/src/main/java/gtfu/GPSData.java:9]: It would be cleaner to set it to 1 and not require count as a parameter.

Comment (unreasonableman) [reply gtfu/src/main/java/gtfu/GPSData.java:9]: Don't know if I'm understanding correctly, but you can overload constructors:
```
public GPSData(long millis, float lat, float lon) {
    this(long millis, float lat, float lon, 1);
}

public GPSData(long millis, float lat, float lon, int count) {
    ...
}
```

File: gtfu/src/main/java/gtfu/tools/DayLogSlicer.java [hunk 2]
```diff
@@ -56,14 +57,21 @@ public DayLogSlicer(TripCollection tripCollection, RouteCollection routeCollecti
...
-                gpsMap.put(tripID, list);
+            String latLon = String.valueOf(lat) + String.valueOf(lon);
+            if(gpsMap.get(tripID) == null){
+                Map<String, GPSData> latLongMap = new HashMap();
+                latLongMap.put(latLon, new GPSData(seconds * 1000l, lat, lon, 1));
+                gpsMap.put(tripID,latLongMap);
+            }
+            else {
+                if (gpsMap.get(tripID).get(latLon) == null) {
+                    gpsMap.get(tripID).put(latLon, new GPSData(seconds * 1000l, lat, lon, 1));
```
Comment (unreasonableman) [inline gtfu/src/main/java/gtfu/tools/DayLogSlicer.java:69]: I don't understand why there are these two very similar if statements here, please add a comment.

**Review by unreasonableman — CHANGES_REQUESTED** [review]
Looks good mostly, add request for explanation for one bit.

**Comment by owades** [comment]
@unreasonableman ready for re-review. You were right about the redundant logic - I cleaned it up, and also added comments.

**Review by unreasonableman — APPROVED** [review]

**Comment by owades** [comment]
@unreasonableman would you prefer I hold on merging until we have addressed needed changes to TrainingDataUtil?

**Comment by unreasonableman** [comment]
Go ahead and merge.

**Pull request merged**